controllers/topology/internal/extensions/patches/inline/json_patch_generator.go

```diff
@@ -132,6 +142,22 @@ func templateMatchesSelector(templateRef *api.TemplateRef, selector clusterv1.Pa
 	}
 }
 
+func patchIsEnabled(enabledIf *string, variables map[string]apiextensionsv1.JSON) (bool, error) {
+	// If enabledIf is not set, patch is enabled.
+	if enabledIf == nil {
+		return true, nil
+	}
+
+	// Rendered template.
+	value, err := renderValueTemplate(*enabledIf, variables)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to calculate value for enabledIf")
+	}
+
+	// Patch is enabled if the rendered template value is `true`.
+	return bytes.Equal(value.Raw, []byte(`true`)), nil
```

Comment:
Can we unmarshal into a boolean instead? Or does `value` already handle cases like there is a leading newline, etc?

Reply:
We are using `yaml.Unmarshal` in `renderValueTemplate` which is able to handle leading and trailing new lines.

I pushed 2 new unit tests to also cover those cases (named something with `(even with leading and trailing new line)`)